src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+    "aria-describedby",
+    `${mockedGuid}-field-help`
+  );
+});
+
+test.each(["info", "warning", "error"])(
+  "with id prop provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(
+      <Textarea
+        id="foo"
+        label="bar"
+        fieldHelp="baz"
+        {...{ [validationType]: "test" }}
+      />
+    );
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(screen.getByRole("tooltip")).toHaveAttribute("id", "foo-validation");
+    expect(textarea).toHaveAttribute(
+      "aria-describedby",
+      "foo-field-help foo-validation"
+    );
+  }
+);
+
+test.each(["info", "warning", "error"])(
+  "with id prop not provided, %s prop set as a string and the textarea element focused, the id of the validation tooltip should be added to 'aria-describedby' in the textarea element",
+  (validationType) => {
+    render(
+      <Textarea label="bar" fieldHelp="baz" {...{ [validationType]: "test" }} />
+    );
+    const textarea = screen.getByRole("textbox");
+    textarea.focus();
+
+    expect(screen.getByRole("tooltip")).toHaveAttribute(
+      "id",
+      `${mockedGuid}-validation`
+    );
+    expect(textarea).toHaveAttribute(
+      "aria-describedby",
+      `${mockedGuid}-field-help ${mockedGuid}-validation`
+    );
+  }
+);
+
+describe(`when the characterLimit prop is passed`, () => {
+  it.each([2, 3, 4])("renders a character counter", (characterLimit) => {
+    const valueString = "foo";
+    const limitMinusValue = characterLimit - valueString.length >= 0;
+    render(<Textarea value={valueString} characterLimit={characterLimit} />);
+    const underCharacters =
+      characterLimit - valueString.length === 1 ? "character" : "characters";
+    const overCharacters =
+      valueString.length - characterLimit === 1 ? "character" : "characters";
+
+    expect(
+      screen.getByText(
+        `${
+          limitMinusValue
+            ? `${characterLimit - valueString.length} ${underCharacters} left`
+            : `${
+                valueString.length - characterLimit
+              } ${overCharacters} too many`
+        }`,
+        { selector: '[aria-hidden="true"]' }
+      )
+    ).toBeInTheDocument();
+  });
+
+  it("renders a visually hidden hint with id generated via guid", () => {
+    render(<Textarea value="foo" characterLimit={73} />);
+    expect(
+      screen.getByText("You can enter up to 73 characters", {
+        selector: '[data-element="visually-hidden-hint"]',
+      })
+    ).toHaveAttribute("id", mockedGuid);
+  });
+
+  it("references the visually hidden hint id in the input's aria-described by", () => {
+    render(<Textarea value="foo" characterLimit={73} />);
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "aria-describedby",
+      mockedGuid
+    );
+  });
```

Comment:
**suggestion**: what do you think about amending the test description to focus on the intended test outcome, rather than the implementation details. This might help the purpose of the tested code clearer to contributors.

```tsx
it("renders character limit information in a visually hidden container", () => {});
it("character limit information forms part of the textarea's accessible description", () => {});
```